src/main/java/net/sf/jabref/importer/fileformat/BibtexParser.java

```diff
@@ -224,10 +224,14 @@ private void parseAndAddEntry(String type) {
          * for the user.
          */
         try {
+            String commentsAndEntryTypeDefinitionString = dumpTextReadSoFarToString();
+
             BibEntry entry = parseEntry(type);
 
+            entry.setParsedSerialization(commentsAndEntryTypeDefinitionString+dumpTextReadSoFarToString());
+            entry.setCommentsBeforeEntry(commentsAndEntryTypeDefinitionString.substring(0,commentsAndEntryTypeDefinitionString.lastIndexOf('@')));
```

Comment:
I would extract the `@` symbol as a constant, to make it clear. e.g.  private final `COMMENT_SYMBOL = '@'`